### PR TITLE
Added Docker support

### PR DESCRIPTION
**Pull request by iamstefin**

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Maybe add a default Volume for the generated certs to not accidentally lose them.

**Comment by IamStefin** [comment]
> Maybe add a default Volume for the generated certs to not accidentally lose them.

seems good 
can you do it!

File: contrib/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+COPY . /mkp224o/
+
+RUN ./autogen.sh \
+  && ./configure \
+  && make \
+  && cp /mkp224o/mkp224o /usr/local/bin/
+
+WORKDIR /root
+ENTRYPOINT ["mkp224o"]
```
Comment (SuperSandro2000) [inline contrib/docker/Dockerfile:16]:
```suggestion
VOLUME /root/data
ENTRYPOINT ["mkp224o", "-d", "/root/data"]
```

File: contrib/docker/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Dockerfile
+.dockerignore
+README.txt
+.git
```
Comment (cathugger) [inline contrib/docker/.dockerignore:4]: Dockerfile and .dockerignore are too small to matter for ignoring purposes imo. Aint .dockerignore itself autoignored?
README.txt is also small though.
My recommendation is to only leave .git.

File: contrib/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+COPY . /mkp224o/
+
+RUN ./autogen.sh \
+  && ./configure \
+  && make \
+  && cp /mkp224o/mkp224o /usr/local/bin/
+
+WORKDIR /root
+VOLUME /root/data
+ENTRYPOINT ["mkp224o", "-d", "/root/data"]
```
Comment (cathugger) [inline contrib/docker/Dockerfile:17]: couldn't this be simplified to
```
VOLUME /root/data
WORKDIR /root/data
ENTRYPOINT ["mkp224o"]
```
?

Comment (IamStefin) [reply contrib/docker/Dockerfile:17]: Yes this can be simplified like this !

File: contrib/docker/.dockerignore (hunk 2 above)
Comment (SuperSandro2000) [reply contrib/docker/.dockerignore:1]: Size does not really matter in this case. It is about cache busting. If I change the README the Docker Image does not need to rebuild.

**Comment by cathugger** [comment]
how to actually run it? id like to try it out.

**Comment by IamStefin** [comment]
Install docker and build the image locally by this command
`docker build -f contrib/docker/Dockerfile -t mkp224o .`

then run the container using 
`docker run -it -v $(pwd):/root/data mkp224o neko`

**Comment by cathugger** [comment]
got it working.
it's quite okay I guess.
I mean it worked for me as described, putting out onions in current directory.
README.txt or .md would be nice because that /root/data param is unobvious.
At least add these 2 command lines as examples. You could basically copypaste previous comment into readme.md and it would be okay.
Also consider adding MAINTAINER line to Dockerfile.

**Comment by SuperSandro2000** [comment]
>  Also consider adding MAINTAINER line to Dockerfile.

Which is deprecated and got replaced with the label maintainer.

**Comment by cathugger** [comment]
> Which is deprecated and got replaced with the label maintainer.

yeah it seems `LABEL maintainer="..."` is preferred now

>  README.txt or .md would be nice because that /root/data param is unobvious.

to be precise, in contrib/docker folder, not main readme

**Comment by iamstefin** [comment]
@cathugger All done!

**Comment by cathugger** [comment]
alright.

**Pull request merged**